File: src/leetcode/java/KdiffPairsInAnArray_532.java
import java.util.HashMap;
import java.util.Map;

public class KdiffPairsInAnArray_532 {
  public int findPairs(int[] nums, int k) {
    if (k < 0) return 0;

    HashMap<Integer, Integer> counter = new HashMap<>();
    for (int n : nums) {
      counter.put(n, counter.getOrDefault(n, 0) + 1);
    }

    int pairs = 0;
    for (Map.Entry<Integer, Integer> e : counter.entrySet()) {
      if (k == 0) {
        if (e.getValue() > 1) pairs++;
      } else {
        if (counter.containsKey(e.getKey() + k)) pairs++;
      }
    }
    return pairs;
  }
}

File: src/leetcode/python/KdiffPairsInAnArray_532.py
import collections

class KdiffPairsInAnArray_532(object):
    def findPairs(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        if k < 0:
            return 0

        cnt = collections.Counter(nums)
        pairs = 0
        for c in cnt:
            if k == 0 and cnt[c] > 1:
                pairs += 1
            elif k != 0 and cnt[c + k]:
                pairs += 1
        return pairs
